tests/decorators/test_decorators.py

```diff
@@ -0,0 +1,802 @@
+                                         "type": "cosmosDB",
+                                         "name": "out",
+                                         "databaseName": "dummy_out_db",
+                                         "collectionName":
+                                             "dummy_out_collection",
+                                         "connectionStringSetting": "dummy_str"
+                                     },
+                                     {
+                                         "direction": BindingDirection.IN,
+                                         "type": "cosmosDB",
+                                         "name": "in",
+                                         "databaseName": "dummy_in_db",
+                                         "collectionName":
+                                             "dummy_in_collection",
+                                         "connectionStringSetting": "dummy_str"
+                                     },
+                                     {
+                                         "direction": BindingDirection.IN,
+                                         "type": "cosmosDBTrigger",
+                                         "name": "trigger",
+                                         "databaseName": "dummy_db",
+                                         "collectionName": "dummy_collection",
+                                         "connectionStringSetting": "dummy_str"
+                                     }
+                                 ]
+                                 })
+
+    def test_multiple_triggers(self):
+        app = self.func_app
+        with self.assertRaises(ValueError) as err:
+            trigger1 = TimerTrigger(name="req1", schedule="dummy_schedule")
+            trigger2 = TimerTrigger(name="req2", schedule="dummy_schedule")
+
+            @app.schedule(arg_name="req1", schedule="dummy_schedule")
+            @app.schedule(arg_name="req2", schedule="dummy_schedule")
+            def dummy():
+                pass
+        self.assertEqual(err.exception.args[0],
+                         "A trigger was already registered to this "
+                         "function. Adding another trigger is not the "
+                         "correct behavior as a function can only have one "
+                         "trigger. Existing registered trigger "
+                         f"is {trigger2.get_dict_repr()} and New trigger "
+                         f"being added is {trigger1.get_dict_repr()}")
+
+    def test_no_trigger(self):
+        app = self.func_app
+        with self.assertRaises(ValueError) as err:
+            @app.write_queue(arg_name="out", queue_name="dummy_out_queue",
+                             connection="dummy_out_conn")
+            def dummy():
+                pass
+
+            app.get_functions()
+
+        self.assertEqual(err.exception.args[0],
+                         "Function dummy does not have a trigger")
```

Comment:
We should make the error explanation a little bigger - and once the documentation is up, ada an aka.ms link